src/mgmt/cli_mgmt.c

```diff
@@ -231,8 +231,8 @@ get_attach_info(const char *name, bool all_ranks, struct dc_mgmt_sys_info *info,
 	rc = drpc_connect(dc_agent_sockpath, &ctx);
 	if (rc != -DER_SUCCESS) {
 		D_ERROR("failed to connect to %s " DF_RC "\n",
-			dc_agent_sockpath, DP_RC(rc));
-		D_GOTO(out, 0);
+			dc_agent_sockpath, DP_RC(-DER_AGENT_COMM));
+		D_GOTO(out, rc = -DER_AGENT_COMM);
```

Comment:
You need to only do this for the specific error that means the agent isn't running.
```suggestion
			dc_agent_sockpath, DP_RC(rc));
		if (rc == -DER_NOENT)
			rc = -DER_AGENT_COM;	
		D_GOTO(out, rc);
```

Reply:
This is how the output looks as stands in the PR:
`daos ERR  src/common/drpc.c:217 unixcomm_connect() Failed to connect to /var/run/daos_agent/daos_agent.sock, errno=2(No such file or directory)
mgmt ERR  src/mgmt/cli_mgmt.c:234 get_attach_info() failed to connect to /var/run/daos_agent/daos_agent.sock DER_AGENT_COMM(-2034): 'Agent communication error'
ERROR: daos: daos_init() failed: DER_AGENT_COMM(-2034): Agent communication error`

That suggestion yields:
`daos pool query --pool=8b15e064-c956-4bfa-ac46-64fb1f147f85
daos ERR  src/common/drpc.c:220 unixcomm_connect() Failed to connect to /var/run/daos_agent/daos_agent.sock, errno=2(No such file or directory)
mgmt ERR  src/mgmt/cli_mgmt.c:234 get_attach_info() failed to connect to /var/run/daos_agent/daos_agent.sock DER_NONEXIST(-1005): 'The specified entity does not exist'
ERROR: daos: daos_init() failed: DER_NONEXIST(-1005): The specified entity does not exist`

Is that the behavior we would like to see?

Reply:
This seems reasonable.

Reply:
Having both -DER_NONEXIST and -DER_AGENT_COMM in the output actually makes the error clearer IMO, however I wasn't making my suggestion to improve the output, but saying that you shouldn't universally change the error code, just the  error codes that you expect for the failure you're trying to handle.  The fault injection tests (rightly) won't pass with the cod as it stands when I made my original comment.

Reply:
Additionally, I don't have strong opinions on which should come first, the assignment or the D_ERROR() call, either would be fine depending on what you want the output to look like.

Reply:
I agreed with the comments a few days back. This is great feedback and I've incorporated these suggestions.